compose/service.py

```diff
@@ -780,14 +780,22 @@ def build_container_name(project, service, number, one_off=False):
 
 # Images
 
-
 def parse_repository_tag(s):
-    if ":" not in s:
-        return s, ""
-    repo, tag = s.rsplit(":", 1)
-    if "/" in tag:
-        return s, ""
-    return repo, tag
+    tag_sep = ":"
+    digest_sep = "@"
+
+    repo, tag, separator = s, "", tag_sep
+
+    if digest_sep in s:
+        separator = digest_sep
+        repo, tag = s.rsplit(digest_sep, 1)
+
+    elif tag_sep in s:
+        repo, tag = s.rsplit(tag_sep, 1)
+        if "/" in tag:
+            repo, tag = s, ""
+
+    return repo, tag, separator
```

Comment:
I'm finding this function hard to parse/has a cognitive load to understand intended behaviour. To help me understand, I re-wrote it.:

`def parse_repository_tag(repo_path):
    tag_separator = ":"
    digest_separator = "@"

    if digest_separator in repo_path:
        repo, tag = repo_path.rsplit(digest_separator, 1)
        return repo, tag, digest_separator

    if tag_separator in repo_path:
        repo, tag = repo_path.rsplit(tag_separator, 1)
        if "/" in tag:
            return repo_path, "", tag_separator

        return repo, tag, tag_separator

    return repo_path, "", tag_separator
`

How do you feel about this way of expressing the function? Using fully expanded variable names really helps with readability.